iaso/models/forms.py

```diff
@@ -87,13 +87,18 @@ def filter_for_user_and_app_id(self, user: typing.Union[User, AnonymousUser], ap
         if app_id is not None:  # mobile app
             try:
                 project = Project.objects.get_for_user_and_app_id(user, app_id)
-                queryset = queryset.filter(projects__in=[project])
-                queryset = queryset.exclude(derived=True)  # do not include derived instances for the mobile app
+                queryset = self.filter_for_project(project=project, queryset=queryset)
             except Project.DoesNotExist:
                 return self.none()
 
         return queryset
 
+    def filter_for_project(self, project: Project, queryset=None):
+        if queryset is None:
+            queryset = self.all()
```

Comment:
I'm not sure if this is actually going to change anything performance-wise, but it should be better

```suggestion
            queryset = self
```